### ISSUE TITLE
Linux IBC Training is Broken

### ISSUE THREAD

**Issue by brianrob** [dotnet/coreclr#24275]
After #24229 was merged, Linux IBC training no longer generates data.  I have confirmed that without this change, IBC data is generated.

Training fails with the following error, which says that no IBC files were found:
```
[12:08:23] Looking for IBC files in directory '/home/brianrob/src/dotnet-optimization/output/CLRx64LINmas/IBC/coreclr-master/bin/Product/Linux.x64.Release'.
[12:08:23] Looking for IBC files in directory '/home/brianrob/src/dotnet-optimization/output/CLRx64LINmas/IBC/cli/x64'.
[12:08:23] Failure during training scenario 'DotNet_TechEmpower'.  Exception details: System.Exception: Failed training for CLRx64LINmas: the last training scenario did not produce any data
   at Microsoft.DotNet.Optimization.Linux_IBC.AcceptTrain(Product product, Scenario scenario) in /home/brianrob/src/dotnet-optimization/src/optimizationtools/Linux_IBC.cs:line 68
   at Microsoft.DotNet.Optimization.TrainingJob.GenerateTrainingData(String layoutDirectory) in /home/brianrob/src/dotnet-optimization/src/core/TrainingJob.cs:line 134

Unhandled Exception: System.Exception: Failed training for CLRx64LINmas: the last training scenario did not produce any data
   at Microsoft.DotNet.Optimization.Linux_IBC.AcceptTrain(Product product, Scenario scenario) in /home/brianrob/src/dotnet-optimization/src/optimizationtools/Linux_IBC.cs:line 68
   at Microsoft.DotNet.Optimization.TrainingJob.GenerateTrainingData(String layoutDirectory) in /home/brianrob/src/dotnet-optimization/src/core/TrainingJob.cs:line 134
   at Microsoft.DotNet.Optimization.TrainingJob.Execute() in /home/brianrob/src/dotnet-optimization/src/core/TrainingJob.cs:line 53
   at Microsoft.DotNet.Optimization.AutomatedOptimizationJob.processJob(Job job, Action`1 onSuccess) in /home/brianrob/src/dotnet-optimization/src/jobs/AutomatedOptimizationJob.cs:line 22
   at Microsoft.DotNet.Optimization.AutomatedOptimizationJob.runJobs() in /home/brianrob/src/dotnet-optimization/src/jobs/AutomatedOptimizationJob.cs:line 131
   at Microsoft.DotNet.Optimization.Program.Main(String[] args) in /home/brianrob/src/dotnet-optimization/src/Program.cs:line 29
./train.sh: line 77:  4998 Aborted                 (core dumped) $dotnetCmd $currentDir/artifacts/bin/optimization/$buildConfiguration/netcoreapp2.0/optimization.dll --output-directory="$outputDirectory" --train $jobs $rerunscenarios
```

**Comment by brianrob** [comment]
@briansull can you please take a look?  This appears to be specific to Linux and is currently blocking publish of all IBC and PGO data.

**Comment by briansull** [comment]
OK, I will take a look.  
I did verify that by hand I was able to generate IBC data and had the scenario named correctly.

**Comment by brianrob** [comment]
Thanks much.  I can also give you instructions on how to run the training system on your own machine to verify that things work too.  You'd just need a fork of CoreCLR to point it at and it would clone, build, deploy, run all on its own.

**Comment by briansull** [comment]
I have a fork on my Linux VM system.

**Comment by brianrob** [comment]
Ok sounds good.  If you have something that you believe works, post a branch to GitHub and I can also run it locally and see if it works in the training system.

**Comment by briansull** [comment]
I don't have your instructions on how to run the training system on my own machine.

**Comment by brianrob** [comment]
I just sent you some mail with instructions.

**Comment by briansull** [comment]
Resolved with #24283
